test/indexed_test.dart

```diff
@@ -0,0 +1,199 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:math';
+import 'dart:typed_data';
+
+import 'package:collection/collection.dart';
+import 'package:test/test.dart';
+
+import '../lib/realm.dart';
+
+part 'indexed_test.g.dart';
+
+// IMPORTANT: Don't import our own test.dart here. It will break AOT compilation!
```

Comment:
Why do we need AOT compilation for these tests? Where do we invoke it?

Reply:
For CI it doesn't matter, but when working on improving performance, it is nice to be able to do
```sh
dart compile exe test/indexed_test.dart
```

Reply:
how can we improve performance here. Isn't it a Core work.  I suggest we don't restrict ourselves with this limitation

Reply:
Yes, this is just a test. It has already exposed that indexes on timestamps are not very beneficial currently.

Reply:
can we at least have a comment why do we need AOT compilation in here. Just so it is more clear what we are trying to achieve. 
Something like 
`We may use AOT compilation locally to run the performance tests in this file`

Reply:
Sure, I'll elaborate on the comment. We should probably also figure out why it breaks AOT compilation of the tests to include our own `test.dart` eventually.

Reply:
```dart
// Don't import our own test.dart here. It will break AOT compilation.
// We may use AOT compilation locally to manually run the performance
// tests in this file
```